gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -338,51 +374,44 @@ public int read(ByteBuffer byteBuffer) throws IOException {
     logger.atFiner().log(
         "GCS gRPC read request for up to %d bytes at offset %d from object '%s'",
         byteBuffer.remaining(), position(), resourceId);
+    metricsPublisher.recordTaggedStat(METHOD, "read", REQUESTS, 1L);
 
     if (!isOpen()) {
       throw new ClosedChannelException();
     }
-    long startTime = System.currentTimeMillis();
-    try {
-      int bytesRead = 0;
-
-      if (resIterator != null && isByteBufferBeyondCurrentRequestRange(byteBuffer)) {
-        positionInGrpcStream += bytesToSkipBeforeReading;
-        cancelCurrentRequest();
-        invalidateBufferedContent();
-        bytesToSkipBeforeReading = 0;
-      }
+    int bytesRead = 0;
 
-      // The server responds in 2MB chunks, but the client can ask for less than that. We
-      // store the remainder in bufferedContent and return pieces of that on the next read call (and
-      // flush that buffer if there is a seek).
-      if (bufferedContent != null) {
-        bytesRead += readBufferedContentInto(byteBuffer);
-      }
-      if (!byteBuffer.hasRemaining()) {
-        return bytesRead;
-      }
-      if (positionInGrpcStream == objectSize) {
-        return bytesRead > 0 ? bytesRead : -1;
-      }
+    if (resIterator != null && isByteBufferBeyondCurrentRequestRange(byteBuffer)) {
+      positionInGrpcStream += bytesToSkipBeforeReading;
+      cancelCurrentRequest();
+      invalidateBufferedContent();
+      bytesToSkipBeforeReading = 0;
+    }
 
-      long effectivePosition = positionInGrpcStream + bytesToSkipBeforeReading;
-      if ((footerBuffer == null) || (effectivePosition < footerStartOffsetInBytes)) {
-        OptionalLong bytesToRead = getBytesToRead(byteBuffer);
-        bytesRead += readFromGCS(byteBuffer, bytesToRead);
-      }
+    // The server responds in 2MB chunks, but the client can ask for less than that. We
+    // store the remainder in bufferedContent and return pieces of that on the next read call (and
+    // flush that buffer if there is a seek).
+    if (bufferedContent != null) {
+      bytesRead += readBufferedContentInto(byteBuffer);
+    }
+    if (!byteBuffer.hasRemaining()) {
+      return bytesRead;
+    }
+    if (positionInGrpcStream == objectSize) {
+      return bytesRead > 0 ? bytesRead : -1;
+    }
 
-      if (hasMoreFooterContentToRead(byteBuffer)) {
-        bytesRead += readFooterContentIntoBuffer(byteBuffer);
-      }
+    long effectivePosition = positionInGrpcStream + bytesToSkipBeforeReading;
+    if ((footerBuffer == null) || (effectivePosition < footerStartOffsetInBytes)) {
+      OptionalLong bytesToRead = getBytesToRead(byteBuffer);
+      bytesRead += readFromGCS(byteBuffer, bytesToRead);
+    }
 
-      return bytesRead;
-    } finally {
-      long endTime = System.currentTimeMillis();
-      logger.atFinest().log(
```

Comment:
Let's keep the log, they are super useful during debugging. metricsPublisher is disabled by default and it only has aggregate metrics published to cloudmonitring.